aptos-move/aptos-workspace-server/src/main.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use aptos::node::local_testnet::HealthChecker;
+use aptos_config::config::NodeConfig;
+use aptos_faucet_core::server::{FunderKeyEnum, RunConfig};
+use aptos_node::{load_node_config, start_and_report_ports};
+use aptos_types::network_address::{NetworkAddress, Protocol};
+use futures::channel::oneshot;
+use rand::{rngs::StdRng, SeedableRng};
+use std::{
+    net::{IpAddr, Ipv4Addr},
+    path::Path,
+    thread,
+    time::Duration,
+};
+use url::Url;
+
+pub fn zero_all_ports(config: &mut NodeConfig) {
+    // TODO: Double check if all ports are covered.
+
+    config.admin_service.port = 0;
+    config.api.address.set_port(0);
+    config.inspection_service.port = 0;
+    config.storage.backup_service_address.set_port(0);
+    config.indexer_grpc.address.set_port(0);
+
+    if let Some(network) = config.validator_network.as_mut() {
+        network.listen_address = NetworkAddress::from_protocols(vec![
+            Protocol::Ip4("0.0.0.0".parse().unwrap()),
+            Protocol::Tcp(0),
+        ])
+        .unwrap();
+    }
+    for network in config.full_node_networks.iter_mut() {
+        network.listen_address = NetworkAddress::from_protocols(vec![
+            Protocol::Ip4("0.0.0.0".parse().unwrap()),
+            Protocol::Tcp(0),
+        ])
+        .unwrap();
+    }
+}
+
+async fn spawn_node(test_dir: &Path) -> Result<()> {
```

Comment:
should we limit the number of ports to spin? to not overload the local machine

Reply:
I don't think there's an effective way to limit the number of ports directly. As I mentioned previously, we're probably better-off limiting the number of concurrent processes spawned. 

Based on my observation, a node currently uses ~10 ports max if we include some temporary ones. We should have no trouble spawning 16 or 32 nodes.